[C++, flatbuffers 1.6+] Passing char * to FlatBufferBuilder::CreateString fails to compile

Since flatbuffers 1.6, it seems that it's no longer possible to pass a `char *` argument to `FlatBufferBuilder::CreateString()` because the wrong overload gets chosen. For example, if I compile the following:

```cpp
#include <flatbuffers/flatbuffers.h>

namespace fb = flatbuffers;

int main()
{
    fb::FlatBufferBuilder builder;
    char buf[] = "hello";
    builder.CreateString(buf);
}
```

I get this (with gcc 7.2.0 + libstdc++):

```
[/tmp] g++ fb.cpp -I /stuff/git/flatbuffers/include
In file included from fb.cpp:1:0:
/stuff/git/flatbuffers/include/flatbuffers/flatbuffers.h: In instantiation of ‘flatbuffers::Offset<flatbuffers::String> flatbuffers::FlatBufferBuilder::CreateString(const T&) [with T = char [6]]’:
fb.cpp:9:29:   required from here
/stuff/git/flatbuffers/include/flatbuffers/flatbuffers.h:1027:29: error: request for member ‘c_str’ in ‘str’, which is of non-class type ‘const char [6]’
     return CreateString(str.c_str(), str.length());
                         ~~~~^~~~~
/stuff/git/flatbuffers/include/flatbuffers/flatbuffers.h:1027:42: error: request for member ‘length’ in ‘str’, which is of non-class type ‘const char [6]’
     return CreateString(str.c_str(), str.length());
                                      ~~~~^~~~~~
```

I did a git bisect and this appears to have been introduced by commit f2071e4f80bf7ccd924337c7bf1bf33c79e3fc3e.

--- a/include/flatbuffers/flatbuffers.h
+++ b/include/flatbuffers/flatbuffers.h
@@ -1001,6 +1001,13 @@ class FlatBufferBuilder {
     return CreateString(str, strlen(str));
   }
 
+  /// @brief Store a string in the buffer, which is null-terminated.
+  /// @param[in] str A char pointer to a C-string to add to the buffer.
+  /// @return Returns the offset in the buffer where the string starts.
+  Offset<String> CreateString(char *str) {
+    return CreateString(str, strlen(str));
+  }
+
   /// @brief Store a string in the buffer, which can contain any binary data.
   /// @param[in] str A const reference to a std::string to store in the buffer.
   /// @return Returns the offset in the buffer where the string starts.

